### PR TITLE
CCUBE-1348

### PR DESCRIPTION
**Pull request by ShIrInRuTh**
**Changes**

- Added a new function to generate id in generate function for element-object-generator
- Added new tests for generate function
- Ammended hook function when adding new elements to include existingElements so that it can be checked when generating a new id.

**Additional information**

-   You may refer to this [CCUBE-1348](https://jira.ship.gov.sg/browse/CCUBE-1348)
- This branch is branched out from CCUBE-1294
- There is only one commit for this the rest is changes merged from main & commits from CCUBE-1294 since its branched from there

### CODE REVIEW

File: src/util/element-object-generator.ts
Comment (weili-govtech) [inline src/util/element-object-generator.ts:1]: - to update with latest changes from main branch
- `generate()` and `duplicate()` have similar logic, to abstract the common parts where possible (DRY principle)

File: src/util/element-object-generator.ts [hunk 1]
```diff
@@ -8,16 +8,56 @@ import { ELEMENT_BUTTON_LABELS } from "src/data";
...
+                    return generateId("numeric-field");
+                case EElementType.CONTACT:
+                    return generateId("contact-field");
+                case EElementType.TEXT:
+                    return generateId("short-text-field");
+                case EElementType.TEXTAREA:
+                    return generateId("long-text-field");
+                default:
+                    return null;
+            }
```
Comment (weili-govtech) [inline src/util/element-object-generator.ts]: create mapping of element type to prefix in `elements-data.ts`

**Review by weili-govtech — CHANGES_REQUESTED** [review]

File: src/util/element-object-generator.ts (hunk 1 above)
Comment (ShIrInRuTh) [reply src/util/element-object-generator.ts]: fixed in latest push

File: src/util/element-object-generator.ts
Comment (ShIrInRuTh) [reply src/util/element-object-generator.ts:1]: fixed in latest push

File: src/util/element-object-generator.ts [hunk 2]
```diff
@@ -36,6 +74,7 @@ export namespace ElementObjectGenerator {
                     conditionalRendering: [],
                     prefill: [],
                 };
+                console.log(attributes);
```
Comment (weili-govtech) [inline src/util/element-object-generator.ts]: remove

**Review by weili-govtech — CHANGES_REQUESTED** [review]

File: src/util/element-object-generator.ts (hunk 2 above)
Comment (ShIrInRuTh) [reply src/util/element-object-generator.ts]: fixed in the last push

**Review by weili-govtech — APPROVED** [review]

**Pull request merged**